keras/engine/training_test.py

```diff
@@ -2390,6 +2390,63 @@ def call(
         self.assertEqual(input_specs[1].shape.as_list(), [2, 2])
         self.assertEqual(input_specs[2].shape.as_list(), [3, 3])
 
+    def test_states(self):
+        inp = keras.Input(batch_shape=[1, 2, 3])
+
+        stateless = keras.layers.SimpleRNN(units=3, stateful=False)
+        stateful = keras.layers.SimpleRNN(units=3, stateful=True)
+
+        model = keras.Model(inp, [stateless(inp), stateful(inp)])
+
+        states = model.states
+        states_ = stateful.states
+        assert len(states) == len(states_)
+        for a, b in zip(states, states_):
+            assert a is b
+
+    def test_get_states(self):
+        inp = keras.Input(batch_shape=[1, 2, 3])
+
+        stateless = keras.layers.SimpleRNN(units=3, stateful=False)
+        stateful = keras.layers.SimpleRNN(units=3, stateful=True)
+
+        model = keras.Model(inp, [stateless(inp), stateful(inp)])
+
+        states = model.get_states()
+        states_ = stateful.get_states()
+        assert len(states) == len(states_)
+        for a, b in zip(states, states_):
+            assert isinstance(a, np.ndarray)
+            assert isinstance(b, np.ndarray)
+            assert (a == b).all()
+
+    def test_reset_states(self):
+        ref_states = [np.random.rand(1, 3).astype(np.float32)]
+
+        inp = keras.Input(batch_shape=[1, 2, 3])
+
+        stateless = keras.layers.SimpleRNN(units=3, stateful=False)
+        stateful = keras.layers.SimpleRNN(units=3, stateful=True)
+
+        model = keras.Model(inp, [stateless(inp), stateful(inp)])
+
+        # test zero init
+        model.reset_states()
+        states = model.get_states()
+        assert len(states) == len(ref_states)
+        for a in states:
+            assert isinstance(a, np.ndarray)
+            assert (a == 0).all()  # init as 0
```

Comment:
Please change to assertAllClose().

Reply:
I think assertEqual works here, as it should be a pure copy operation.